modules/tinymce/src/core/main/ts/api/Settings.ts

```diff
@@ -129,6 +129,48 @@ const getPlaceholder = (editor: Editor): string =>
   // Fallback to the original elements placeholder if not set in the settings
   editor.getParam('placeholder', DOM.getAttrib(editor.getElement(), 'placeholder'), 'string');
 
+const getEventRoot = (editor: Editor): string => editor.getParam('event_root');
```

Comment:
Each one of these calls is going to create a lambda expression in the JS code output like this:

```
function getEventRoot(editor) {
  return editor.getParam('event_root');
}
```

When minifying, `getParam` won't name-mangle, and we have the keyword `function` in there. You'll get something like:

```
function x(e) { return e.getParam('event_root'); }
```

We could refactor like this:

```
const getter = <K extends keyof EditorSettings (message: string) => (editor: Editor): EditorSettings[K] =>
  editor.getParam(message);

const getEventRoot = getter('event_root');
```

Then the `getEventRoot` will minify like this:

```
var x=g('event_root')
```

Reply:
This is low priority in the grand scheme of things, I've just been spending too much time working on code output size, so I noticed it :)

Reply:
Perhaps we could make an issue for it? It seems that, if we wish to add this pattern, we'd want to add it to all Settings-files, not just the few functions I added.